src/lib/utils/package-manifest.ts

```diff
@@ -0,0 +1,193 @@
+// Utilities to support the inspection of node package "manifests" (package.json's)
+
+import { existsSync, statSync } from "fs";
+import glob = require("glob");
+import { dirname, resolve } from "path";
+import { flatMap } from "./array";
+
+import { readFile } from "./fs";
+import { Logger } from "./loggers";
+
+/**
+ * Helper for the TS type system to understand hasOwnProperty
+ * and narrow a type appropriately.
+ * @param obj the receiver of the hasOwnProperty method call
+ * @param prop the property to test for
+ */
+function hasOwnProperty<X extends {}, Y extends PropertyKey>(
```

Comment:
This signature is unnecessarily complicated. TypeScript already knows that `obj` is of type `X`. Type guard functions will already intersect the return checked type and the known type.

```ts
function hasOwnProperty<K extends PropertyKey>(obj: object, prop: K): obj is Record<K, unknown>
```

Reply:
Thanks!